pkg/specgen/generate/container_create.go

```diff
@@ -21,6 +21,27 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
+func platformToCrunVariant(arch string, os string) string {
+	variant := "crun"
+	if arch == "wasm32" || arch == "wasm64" || os == "wasi" {
+		variant = "crun-wasm"
+	}
+	return variant
+}
+
+func modifyCrunIfNeeded(conf libpod.OCIRuntime, variant string) {
+	if variant != "crun" {
+		if strings.HasSuffix(conf.Path(), "crun") {
+			name := conf.Name()
+			path := conf.Path()
+			nameReplace := strings.ReplaceAll(name, "crun", variant)
+			pathReplace := strings.ReplaceAll(path, "crun", variant)
+			conf.UpdateName(nameReplace)
```

Comment:
We should absolutely not be modifying internal Libpod components from outside Libpod, so this needs to move (or, as I suggested, make a new instance of the OCI runtime to handle WASM)

Reply:
I like the idea of an separate OCI Runtime.